README.md

```diff
@@ -492,6 +495,92 @@ status:
+3. The `env` field in the application resource **MUST** conform to the [EnvVar][env] API.
+4. The `containers` field in the application resource **MUST** conform to the API schema given here:
+
+    ```yaml
+    schema:
+      openAPIV3Schema:
+        properties:
+          containers:
+            description: Containers describes which containers in a Pod should be bound to
+            items:
+              anyOf:
+              - type: integer
+              - type: string
+              x-kubernetes-int-or-string: true
+            type: array
+    ```
+
+A `BindingPathMap` resource **MUST** define a `.status.conditions` which is an array of `Condition` objects.  A `Condition` object **MUST** define `type`, `status`, and `lastTransitionTime` entries.  At least one condition containing a `type` of `Ready` **MUST** be defined.  The `status` of the `Ready` condition **MUST** have a value of `True`, `False`, or `Unknown`.  The `lastTransitionTime` **MUST** contain the last time that the condition transitioned from one status to another.  A `BindingPathMap` resource **MAY** define `reason` and `message` entries to describe the last `status` transition.  There could be multiple application resources matching the `apiVersion` and `kind`.  Partial failures **MUST** be aggregated and reported on the binding status's `Ready` condition.
+
+The reconciler **MUST** set the `BindingPathMap`'s `Ready` condition according to the same rules set in [Ready Condition Status](#ready-condition-status) for `ServiceBinding` resource.
+
+[cbp-crd]: internal.service.binding_bindingpathmaps.yaml
+[volume]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volume-v1-core
+[volumemount]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volumemount-v1-core
+[env]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#envvar-v1-core
+
+### Resource Type Schema
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: BindingPathMap
+metadata:
+  name:                 # string
+  generation:           # int64, defined by the Kubernetes control plane
+  ...
+spec:
+  application:          # ObjectReference-like
+    apiVersion:         # string
+    kind:               # string
+
+  bindingPaths:
+    volume:             # string, optional
+    volumeMount:        # string, optional, volume must be specified along with this
+    env:                # string, optional
+    containers:         # string, optional
+
+status:
+  conditions:           # []Condition containing at least one entry for `Ready`
+  - type:               # string
+    status:             # string
+    lastTransitionTime: # Time
+    reason:             # string
+    message:            # string
+  observedGeneration:   # int64
+```
+
+### Custom Binding Paths Example
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: BindingPathMap
+metadata:
+  name: v1alpha1.exampleappbuilders.apps.examples.org
```

Comment:
My suggestion around using a deterministic name was that this format should be codified and enforced in the spec.

Reply:
I am not sure how that is going to help. Please explain.

Reply:
You see this pattern frequently for resources that apply to a specific type. For example, the name of `CustomResoruceDefinition` and `APIService` resources is strictly defined, any attempt to use a different name is rejected. This has a couple key advantages:

1. you'll never have conflicting mappings trying to advise the same resource
2. the mappings for a particular resource can be looked up in constant time rather than needing to scan all mapping looking for a match. You can also know when a mapping is not defined.

Strictly speaking it's not required to make this extension implementable, but it makes the implementation much easier, efficient and predictable.